src/Razor/Microsoft.NET.Sdk.Razor/src/GenerateServiceWorkerAssetsManifest.cs

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Runtime.Serialization.Json;
+using System.Security.Cryptography;
+using System.Text;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.AspNetCore.Razor.Tasks
+{
+    public partial class GenerateServiceWorkerAssetsManifest : Task
```

Comment:
Similar to boot.json, we operate on a different set of inputs during build and publish. This task has been modified to do what was previously done across multiple msbuild targets. I need to spend a little bit of time to verify, but it should be just as incremental as before.

Reply:
That's great. Do you feel the test coverage we have for this is adequate? We've had multiple regressions related to SWAM and build/publish/incrementalism, perhaps because it's pretty hard to tell whether it's right just by casually running an app.

Reply:
I retained all the tests from before. My plan's to go over these again and see if there are gaps and see if we could have more combination of tests (VS-style publish + build incrementalism for instance)